packages/dataviz/src/render-widget.tsx

```diff
@@ -45,6 +46,7 @@ type Options = {
   scaleSettings: Partial<ScaleSettings>;
   componentSettings?: ComponentSettings;
   widgetSettings?: WidgetSettings;
+  legend?: LegendSettings;
```

Comment:
Based on that definition - the renderWidget function could receive only generic `LegendSettings`. But it's not true as `BubbleLegendSettings` could also be provided as argument. Please cover that case here.